e2e/next-sandbox/test/multi.test.ts

```diff
@@ -0,0 +1,228 @@
+import type { Page } from "@playwright/test";
+import { test } from "@playwright/test";
+
+import {
+  expectJson,
+  genRoomId,
+  getJson,
+  pickFrom,
+  preparePages,
+  waitForJson,
+} from "./utils";
+
+// NOTE: The tests below don't play well with concurrency just yet. The reason
+// is that they unmount, remount, and then check that the connection ID
+// increased by exactly one. This is not super robust. What matters is that the
+// connection ID increased, but if it increased by more than 1, that could also
+// be just fine. Better to express the conditional like that. For now, just not
+// run them concurrently to avoid it.
+// test.describe.configure({ mode: "parallel" });
+
+const TEST_URL = "http://localhost:3007/multi/";
+
+test.describe("Multiple rooms", () => {
+  let pages: Page[];
+
+  test.beforeEach(async ({}, testInfo) => {
+    const room = genRoomId(testInfo);
+    pages = await preparePages(
+      `${TEST_URL}?room=${encodeURIComponent(room)}`,
+      { n: 1, width: 1024 } // Open only a single, but wider, browser window
+    );
+  });
+
+  test.afterEach(() =>
+    // Close all pages
+    Promise.all(pages.map((page) => page.close()))
+  );
+
+  test("mount, unmount, connection ID must be incremented", async () => {
+    const page = pages[0];
+
+    await page.fill("#input_1", "e2e:multi-A");
+    await page.click("#mount_1");
+
+    await waitForJson(page, "#socketStatus_1", "connected");
+    const connId = (await getJson(page, "#connectionId_1")) as number;
+
+    await page.click("#unmount_1");
+    await page.click("#mount_1");
+    await waitForJson(page, "#socketStatus_1", "connected");
+
+    await expectJson(page, "#connectionId_1", connId + 1);
```

Comment:
Yeah, this would work in the "expectJson" cases, but it would not work in the "waitForJson" cases, where I would also would like to add the ability to provide a predicate function, where checks like that would be much easier to express.

Imagine:

```tsx
await expectJson(page, "#connectionId_1", (value) => value > connId);
await waitForJson(page, "#connectionId_1", (value) => value > connId);
```

This is something I want to implement, but haven't done yet. I'd prefer to keep this out of this PR, but cycle back to implementing that and rewriting these expressions everywhere!

Reply:
Tracking this in #1239.